### PR TITLE
Improve UI for scan command

### PR DESCRIPTION
**Pull request by sarao1310**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Let me review this PR when all tests are green,

File: xray/formats/table.go [hunk 1]
```diff
@@ -17,6 +17,18 @@ type VulnerabilityTableRow struct {
 	IssueId                   string                       `col-name:"Issue ID" extended:"true"`
 }
 
+type VulnerabilityScanTableRow struct {
+	Severity               string                   `col-name:"Severity"`
+	SeverityNumValue       int                      // For sorting
```
Comment (eyalbe4) [inline xray/formats/table.go:22]: The following applies to other places in the code as well - 
Description comments should be placed above the code they describe.

File: xray/utils/resultstable.go [hunk 2]
```diff
@@ -28,23 +31,37 @@ const (
 // In case multipleRoots is true, the field Component will show the root of each impact path, otherwise it will show the root's child.
 // In case one (or more) of the violations contains the field FailBuild set to true, CliError with exit code 3 will be returned.
 // Set printExtended to true to print fields with 'extended' tag.
-func PrintViolationsTable(violations []services.Violation, multipleRoots, printExtended bool) error {
+func PrintViolationsTable(violations []services.Violation, multipleRoots, printExtended, isScan bool) error {
```
Comment (eyalbe4) [inline xray/utils/resultstable.go:34]: The below a[[ly also for the `PrintVulnerabilitiesTable`, `PrintScanResults` and `PrintLicensesTable` functions - 

1. isScan --> scan
2. Let's describe the arguments this function receieves, and specifically the new `scan` argument.

File: xray/utils/resultstable.go [hunk 3]
```diff
@@ -152,12 +169,18 @@ func prepareViolations(violations []services.Violation, multipleRoots, isTable,
 // Set multipleRoots to true in case the given vulnerabilities array contains (or may contain) results of several projects or files (like in binary scan).
 // In case multipleRoots is true, the field Component will show the root of each impact path, otherwise it will show the root's child.
 // Set printExtended to true to print fields with 'extended' tag.
-func PrintVulnerabilitiesTable(vulnerabilities []services.Vulnerability, multipleRoots, printExtended bool) error {
+func PrintVulnerabilitiesTable(vulnerabilities []services.Vulnerability, multipleRoots, printExtended, isScan bool) error {
+	log.Output(noContextMessage + "Below are all vulnerabilities detected.")
```
Comment (eyalbe4) [inline xray/utils/resultstable.go:173]: I suggest removing this line.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I'd be happy to review this PR again following your changes.

File: xray/utils/resultstable.go (hunk 3 above)
Comment (sverdlov93) [reply xray/utils/resultstable.go:173]: I already removed it from dev a week ago together with noContextMessage.

File: xray/formats/table.go (hunk 1 above)
Comment (sarao1310) [reply xray/formats/table.go:22]: Done!

File: xray/utils/resultstable.go (hunk 3 above)
Comment (sarao1310) [reply xray/utils/resultstable.go:173]: Done!

File: xray/formats/conversion.go [hunk 4]
```diff
@@ -81,6 +148,16 @@ func ConvertToComponentTableRow(rows []ComponentRow) (tableRows []DirectDependen
 	return
 }
 
+func ConvertToComponentScanTableRow(rows []ComponentRow) (tableRows []DirectPackagesTableRow) {
```
Comment (eyalbe4) [inline xray/formats/conversion.go:151]: It looks like this function can be made private.

File: xray/formats/table.go [hunk 5]
```diff
@@ -52,11 +87,34 @@ type OperationalRiskViolationTableRow struct {
...
+	NewerVersions          string                   `col-name:"Newer\nVersions" extended:"true"`
+	LatestVersion          string                   `col-name:"Latest\nVersion" extended:"true"`
+}
+
 type DirectDependenciesTableRow struct {
 	Name    string `col-name:"Direct\nDependency"`
 	Version string `col-name:"Direct\nDependency\nVersion"`
 }
 
+type DirectPackagesTableRow struct {
```
Comment (eyalbe4) [inline xray/formats/table.go:113]: This struct and possibly other structs in this file can be made private. Let's check all of them.
The same goes for the fields on those structs.

Comment (sarao1310) [reply xray/formats/table.go:113]: Done

File: xray/utils/resultstable.go [hunk 6]
```diff
@@ -28,23 +28,37 @@ const (
 // In case multipleRoots is true, the field Component will show the root of each impact path, otherwise it will show the root's child.
 // In case one (or more) of the violations contains the field FailBuild set to true, CliError with exit code 3 will be returned.
 // Set printExtended to true to print fields with 'extended' tag.
-func PrintViolationsTable(violations []services.Violation, multipleRoots, printExtended bool) error {
+// If Scan input is true we want to print the scan tables.
```
[the remark this answers is not in the document]
Comment (sarao1310) [reply xray/utils/resultstable.go:31]: Done!

File: xray/utils/resultstable.go [hunk 7]
```diff
@@ -152,12 +166,16 @@ func prepareViolations(violations []services.Violation, multipleRoots, isTable,
 // Set multipleRoots to true in case the given vulnerabilities array contains (or may contain) results of several projects or files (like in binary scan).
 // In case multipleRoots is true, the field Component will show the root of each impact path, otherwise it will show the root's child.
 // Set printExtended to true to print fields with 'extended' tag.
-func PrintVulnerabilitiesTable(vulnerabilities []services.Vulnerability, multipleRoots, printExtended bool) error {
+func PrintVulnerabilitiesTable(vulnerabilities []services.Vulnerability, multipleRoots, printExtended, isScan bool) error {
```
[the remark this answers is not in the document]
Comment (sarao1310) [reply xray/utils/resultstable.go:169]: Done!

File: xray/formats/conversion.go (hunk 4 above)
Comment (sarao1310) [reply xray/formats/conversion.go]: Done

File: xray/formats/table.go [hunk 8]
```diff
@@ -5,40 +5,75 @@ package formats
...
 	ImpactedDependencyVersion string                       `col-name:"Impacted\nDependency\nVersion"`
 	ImpactedDependencyType    string                       `col-name:"Type"`
 }
 
-type OperationalRiskViolationTableRow struct {
-	Severity                  string                       `col-name:"Severity"`
-	SeverityNumValue          int                          // For sorting
-	DirectDependencies        []DirectDependenciesTableRow `embed-table:"true"`
+type licenseViolationScanTableRow struct {
+	LicenseKey string `col-name:"License"`
```
Comment (eyalbe4) [inline xray/formats/table.go:62]: It looks like `LicenseKey` can also be made private. I suggest we go over all the other fields in this file and checl wether they can be made private as well.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: xray/formats/conversion.go [hunk 9]
```diff
@@ -4,86 +4,163 @@ import (
...
-			ImpactedDependencyType:    rows[i].ImpactedDependencyType,
-			DirectDependencies:        ConvertToComponentTableRow(rows[i].Components),
-			IsEol:                     rows[i].IsEol,
+			severityNumValue:          rows[i].SeverityNumValue,
+			impactedDependencyName:    rows[i].ImpactedDependencyName,
+			impactedDependencyVersion: rows[i].ImpactedDependencyVersion,
+			impactedDependencyType:    rows[i].ImpactedDependencyType,
+			directDependencies:        convertToComponentTableRow(rows[i].Components),
+			isEol:                     rows[i].IsEol,
 			Cadence:                   rows[i].Cadence,
```
Comment (eyalbe4) [inline xray/formats/conversion.go:107]: Must this field remain public? Can you check the other fields?

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Let me review this again following your changes.

**Pull request merged**